spec/svatok_codebreaker/game_spec.rb

```diff
@@ -0,0 +1,174 @@
+        game.command_save
+        saved_data = IO.readlines(game.instance_variable_get(:@file_path)).last
+        expect(saved_data.gsub("\n", '')).to eq(game_data)
+      end
+      it 'system message to the command Save' do
+        expect(game.command_save).to eq('----------Score saved!---------')
+      end
+    end
+
+    context '#menu_command' do
+      it 'show message if user enter not valid command' do
+        expect(game.menu_command('sss')).to eq('You entered is not correct command. Enter the correct command.')
+      end
+    end
+
+    context '#valid_guess?' do
+      it 'not valid if guess empty' do
+        expect(game.valid_guess?('')).to eq(false)
+      end
+      it 'not valid if guess is not 4 numbers' do
+        expect(game.valid_guess?('12345')).to eq(false)
+      end
+      it 'not valid if guess is not with numbers from 1 to 6' do
+        expect(game.valid_guess?('7777')).to eq(false)
+      end
+      it 'valid guess' do
+        expect(game.valid_guess?('1234')).to eq(true)
+      end
+    end
+
+    context '#submit_guess' do
+      it 'codebreaker use one attempt' do
+        attempts_old = game.instance_variable_get(:@attempts)
+        game.submit_guess('1234')
+        expect(game.instance_variable_get(:@attempts)).to eq(attempts_old - 1)
+      end
+      it 'system answer by marking the guess' do
+        game.submit_guess('1234')
+        expect(game.instance_variable_get(:@marking_guess)).not_to be_nil
+      end
+      it 'system answer if proposed guess is not valid' do
+        expect(game.submit_guess('11111')).to eq('Your code is not valid!')
+      end
+    end
+
+    context '#result_after_marking' do
+      it 'show marked code' do
+        game.submit_guess('5555')
+        expected_message = game.result_after_marking[:marking_guess]
+        expect(expected_message).to match(/[-+]{1,4}/) | eq('Sorry, but there is no match...')
+      end
+      it 'show count of attempts' do
+        game.submit_guess('5555')
+        expected_message = game.result_after_marking[:attempts]
+        expect(expected_message).to eq('You have 9 attempts.')
+      end
+
+      before do
+        game.instance_variable_set(:@secret_code, 1234)
+      end
+
+      it 'show codebreaker wins' do
+        game.submit_guess('1234')
+        expect(game.result_after_marking[:game_end]).to eq('----------Congratulations! You win!----------')
+      end
+      it 'show codebreaker loses' do
+        game.submit_guess('5555')
+        game.instance_variable_set(:@attempts, 0)
+        expect(game.result_after_marking[:game_end]).to eq('----------Sorry, but you lose :(----------')
+      end
+      it 'show secret_code if loses' do
+        game.submit_guess('5555')
+        game.instance_variable_set(:@attempts, 0)
+        expect(game.result_after_marking[:secret_code]).to eq('Secret code is ' + game.instance_variable_get(:@secret_code).to_s)
+      end
```

Comment:
it is better to use describe/context here to group tests with the same preconditions